base/applications/mspaint/canvas.cpp

```diff
@@ -142,31 +158,204 @@ LRESULT CCanvasWindow::OnVScroll(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL&
     }
     SetScrollInfo(SB_VERT, &si);
     Update(m_hWnd);
+    Invalidate(FALSE); // FIXME: Flicker
     return 0;
 }
 
 LRESULT CCanvasWindow::OnLButtonDown(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
 {
-    switch (toolsModel.GetActiveTool())
+    POINT pt = { GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam) };
+    SIZEBOX_HITTEST sht = HitTest(pt);
+
+    if (sht == SIZEBOX_NONE)
     {
-        case TOOL_BEZIER:
-        case TOOL_SHAPE:
+        if (toolsModel.GetActiveTool() == TOOL_BEZIER ||
+            toolsModel.GetActiveTool() == TOOL_SHAPE)
+        {
             if (ToolBase::pointSP != 0)
             {
                 toolsModel.OnCancelDraw();
                 imageArea.Invalidate();
             }
+        }
+
+        toolsModel.resetTool();  // resets the point-buffer of the polygon and bezier functions
+        return 0;
+    }
+
+    if (sht == SIZEBOX_CONTENTS)
+    {
+        // TODO:
```

Comment:
What symptom/bug is to be expected until that TODO will be addressed?

Reply:
```txt
// TODO: In future, we handle the events of the window-less image area.
```